docs/ContainerRuntimeConfigDesign.md

```diff
@@ -0,0 +1,95 @@
+# Summary
+
+Users need a way to update the container runtime configuration. Users can do this via MCO currently, but they need to know the correct options to use. We can hopefully help customers by documenting a CRD with the ContainerRuntime tuneables and have the MCO use these tuneables when rendering the crio.conf, storage.conf, and registries.conf files to Ignition/disk.
+
+The CRI-O runtime is used by default and is the recommended solution.
+
+# Motivation
+
+The MCO contains the logic to write the crio.conf, storage.conf, and registries.conf configuration files to Ignition. When Ignition starts on a machine it writes these three files to configure the container runtime. When these resources change within the MachineConfig, the new configs are rewritten and the MachineConfigDaemon is instructed to reboot the machine to pick up the new configs.
+
+## Goals
+
+The following goals should not require explicit knowledge of a customer to know the right command-line or configuration options to use within the ContainerRuntime.
+
+1. setting pids limit
+
+2. setting log level
+
+3. configuring list of search registries
+
+4. configuring list of insecure registries
+
+5. configuring list of blocked registries
```

Comment:
How does this overlap with the existing global config fields for this?

Reply:
The fields right now are scoped to build time behavior and not runtime behavior.  It’s possible we should collapse to a common config for both and have MCO render from that global config.  I could see a benefit in letting us configure this at pool level.  In particular, we could have a whitelist registry setting that says only images that run on control planes should come from this source.